src/utils/validation_utils.py

```diff
@@ -45,7 +49,14 @@ def validate_dataset(
             f"Data does not have correct number of metadata fields: {len(da.attrs)}"
         )
         return False
-
+    if (date.day != da.attrs["date_valid"] or
```

Comment:
We may need to adjust this as it's going to throw errors for the other pipelines that don't have `date_valid` stored in their metadata (as they're monthly datasets).

Reply:
See suggested change here: https://github.com/OCHA-DAP/ds-raster-pipelines/pull/50